lib/beautiful/init.lua

```diff
@@ -117,14 +117,25 @@ function beautiful.init(config)
         if type(config) == 'string' then
             -- Expand the '~' $HOME shortcut
             config = config:gsub("^~/", homedir .. "/")
+            configdir = config:gsub("/[^/]+$", "")
```

Comment:
@Elv13 My bad, fix it now.

@psychon Maybe it's them problem of your custom `rc.lua` file, since it works well for me.

My desktop entry file.

```
% cat /usr/share/xsessions/awesome-test.desktop 
[Desktop Entry]
Encoding=UTF-8
Name=awesome-test
Comment=Highly configurable framework window manager
Exec=/usr/local/bin/awesome -c /usr/local/etc/xdg/awesome/rc.lua
Type=Application
```